src/Illuminate/Filesystem/FilesystemAdapter.php

```diff
@@ -273,18 +273,10 @@ public function deleteDirectory($directory)
 	 */
 	protected function filterContentsByType($contents, $type)
 	{
-		$contents = Collection::make($contents);
-
-		$contents = $contents->filter(function($value) use ($type)
-		{
-			return $value['type'] == $type;
-		})
-		->map(function($value)
-		{
-			return $value['path'];
-		});
-
-		return $contents->values()->all();
+		return Collection::make($contents)
+			->where('type', $type)
+			->fetch('path')
+			->values()->all();
```

Comment:
What's the point of `values()`?

`fetch` itself calls `array_values` on its result.